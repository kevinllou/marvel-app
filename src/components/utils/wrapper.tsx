import { render } from '@testing-library/react';
import { ReactElement } from 'react';
import { BrowserRouter, MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import { store } from '../../redux/store';
import MarvelRoutes from '../../routes/MarvelRoutes';

const renderWithBrowserRouter = (component: ReactElement) => render(
  <Provider store={store}>
    <BrowserRouter>{component}</BrowserRouter>
  </Provider>,
);

const renderWithMemoryRouter = (path: string[] = ['/']) => render(
  <Provider store={store}>
    <MemoryRouter initialEntries={path}>
      <MarvelRoutes />
    </MemoryRouter>
  </Provider>,
);

export {
  renderWithBrowserRouter,
  renderWithMemoryRouter,
};
