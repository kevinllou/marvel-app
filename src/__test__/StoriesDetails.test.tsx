import { fireEvent, screen, waitFor } from '@testing-library/react';
import { renderWithMemoryRouter } from '../components/utils/wrapper';

describe('Stories details page', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });
  it('should render details component', async () => {
    const { container } = renderWithMemoryRouter(['/stories/2326']);
    await waitFor(() => {
      expect(container.querySelector('.spinner')).not.toBeTruthy();
      expect(screen.getByText(/Jennifer Walters seeks the one person on Earth who can help control her transformations/i)).toBeInTheDocument();
      expect(screen.getByText('Like')).toBeInTheDocument();
      expect(container).toMatchSnapshot();
    });
  });
  it('should add the stories to favorites', async () => {
    const { container } = renderWithMemoryRouter(['/stories/2326']);
    await waitFor(() => {
      expect(container.querySelector('.spinner')).not.toBeTruthy();
    });
    const likeButton = screen.queryByTestId('icon-heart');
    expect(likeButton?.style.backgroundColor).toBe('red');
    if (likeButton) {
      fireEvent.click(likeButton, { click: 0 });
    }
    expect(likeButton?.style.backgroundColor).toBe('grey');
  });
  test('should hide the story', async () => {
    const { container } = renderWithMemoryRouter(['/stories/2326']);

    await waitFor(() => {
      expect(container.querySelector('.spinner')).not.toBeTruthy();
    });

    const hideButton = screen.queryByTestId('icon-hide');
    expect(hideButton?.style.backgroundColor).toBe('red');
    if (hideButton) {
      fireEvent.click(hideButton, { click: 0 });
    }
    expect(hideButton?.style.backgroundColor).toBe('grey');
  });
});
