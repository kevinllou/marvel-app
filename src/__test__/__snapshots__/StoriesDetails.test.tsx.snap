// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Stories details page should render details component 1`] = `
<div>
  <div
    class="App"
  >
    <section
      class="header"
    >
      <div
        class="header__container"
      >
        <div
          class="header__logo"
        >
          <img
            alt="marvel logo"
            class="header__bgImage"
            src="marvel-icon.png"
          />
        </div>
        <nav
          class="header__navigation"
        >
          <ul
            class="header__navList"
          >
            <li
              class="header__navItems"
            >
              <a
                class=""
                href="/"
              >
                Home
              </a>
            </li>
            <li
              class="header__navItems"
            >
              <a
                class=""
                href="/characters"
              >
                Characters
              </a>
            </li>
            <li
              class="header__navItems"
            >
              <a
                class=""
                href="/comics"
              >
                Comics
              </a>
            </li>
            <li
              class="header__navItems"
            >
              <a
                aria-current="page"
                class="header__activeLink"
                href="/stories"
              >
                Stories
              </a>
            </li>
            <li
              class="header__navItems"
            >
              <a
                class=""
                href="/bookmarks"
              >
                Bookmarks
              </a>
            </li>
          </ul>
        </nav>
        <div
          class="header__menuIcon"
          role="menu"
          tabindex="0"
        >
          <i
            class="fa-solid fa-bars"
          />
        </div>
      </div>
    </section>
    <main>
      <section
        class="detail"
      >
        <div
          class="detail__container"
        >
          <div
            class="detail__goBack"
          >
            <button
              type="button"
            >
              <i
                class="fa-solid fa-arrow-left"
              />
            </button>
          </div>
          <div
            class="detail__card"
          >
            <div
              class="detail__cardImg"
            >
              <img
                alt=""
                src="http://i.annihil.us/u/prod/marvel/i/mg/b/40/image_not_available.jpg"
              />
            </div>
            <div
              class="detail__cardBody"
            >
              <div
                class="detail__cardBodyInfo"
              >
                <p
                  class="detail__cardBodyTitle"
                >
                  Avengers (1998) #74
                </p>
                <p
                  class="detail__cardDescription"
                >
                  "SEARCH FOR SHE-HULK" While the Avengers continue to search the globe for her, Jennifer Walters seeks the one person on Earth who can help control her transformations. Featuring the return of the most demanded Avenger of all! 32 PGS.MARVEL PSR...$2.25
                </p>
              </div>
              <div
                class="detail__cardBodyBottom"
              >
                <button
                  data-testid="icon-heart"
                  style="background-color: red;"
                  type="button"
                >
                  Like
                   
                  <i
                    class="fa-sharp fa-solid fa-heart"
                  />
                </button>
                <button
                  data-testid="icon-hide"
                  style="background-color: red; pointer-events: auto;"
                  type="button"
                >
                  Hide
                   
                  <i
                    class="fa-solid fa-eye-slash"
                  />
                </button>
              </div>
            </div>
          </div>
          <div
            class="detail__characterInfo"
          >
            <div
              class="detail__characters"
            >
              <h1>
                CHARACTERS
              </h1>
              <a
                href="/characters/1010801"
              >
                <li>
                  Ant-Man (Scott Lang)
                </li>
              </a>
              <a
                href="/characters/1009165"
              >
                <li>
                  Avengers
                </li>
              </a>
              <a
                href="/characters/1009220"
              >
                <li>
                  Captain America
                </li>
              </a>
              <a
                href="/characters/1009338"
              >
                <li>
                  Hawkeye
                </li>
              </a>
              <a
                href="/characters/1009368"
              >
                <li>
                  Iron Man
                </li>
              </a>
              <a
                href="/characters/1009562"
              >
                <li>
                  Scarlet Witch
                </li>
              </a>
              <a
                href="/characters/1009583"
              >
                <li>
                  She-Hulk (Jennifer Walters)
                </li>
              </a>
            </div>
            <div
              class="detail__comics"
            >
              <h1>
                COMICS
              </h1>
              <a
                href="/comics/348"
              >
                <li>
                  Avengers (1998) #74
                </li>
              </a>
            </div>
          </div>
        </div>
      </section>
    </main>
    <footer
      class="footer"
    >
      <div
        class="footer__container"
      >
        <div
          class="footer__image"
        >
          <img
            alt="marvel logo"
            src="marvel-icon.png"
          />
        </div>
        <p
          class="footer__copyright"
        >
          Made by KevinðŸ˜Ž. Applaudo Studios 2023
        </p>
      </div>
    </footer>
  </div>
</div>
`;
